src/components/CouchView.svelte

```diff
@@ -2,68 +2,60 @@
   import { onMount } from "svelte";
   import { state as authState } from "../auth.js";
   import { view, all_docs } from "../couch.js";
+  import CouchViewTable, { data } from "../components/CouchViewTable.svelte";
+  import SelectedView, { result } from "../components/SelectedView.svelte";
+
+  import { createEventDispatcher } from "svelte";
+  const dispatch = createEventDispatcher();
 
   export let dbname, ddoc, viewname, options;
   export let viewContents = [];
   export let group_level = 1;
-  export let startkey;
-  export let endkey;
-  export let endkeyIn;
   export let selectedVal = [];
+  let response = [];
 
+  let showResult = true;
   $: token = $authState.token;
   onMount(async () => {
+    await retrieveKey();
+  });
+
+  async function retrieveKey(event) {
     try {
-      let response = await view(token, dbname, ddoc, viewname, options);
+      response = await view(token, dbname, ddoc, viewname, options);
+
       viewContents = response.rows;
     } catch (err) {
       viewContents = [err];
     }
-  });
+  }
 
-  function reduceView() {
-    var updateKey;
-    var tabRecords = document.getElementById("table");
-    var rows = tabRecords.rows;
-    for (var i = 0; i < rows.length; i++) {
-      rows[i].onclick = function() {
-        if (this.parentNode.nodeName == "THEAD") {
-          return;
-        }
-        var rowNumber = this.rowIndex;
-        if (Array.isArray(viewContents[rowNumber - 1].key)) {
-          startkey = JSON.stringify(viewContents[rowNumber - 1].key);
-          viewContents.forEach(item => {
-            (endkeyIn = item.key), item.key.push({});
-            endkey = JSON.stringify(endkeyIn);
-          });
+  function selectKey(event) {
```

Comment:
I think this needs to trigger when the `group_level` value changes.

Reply:
Question about this: If the change of group_level must trigger the function, what about the key selection? What is the expected behaviour of that?